internal/ent/schema/custom_domain.go

```diff
@@ -0,0 +1,148 @@
+package schema
+
+import (
+	"entgo.io/contrib/entgql"
+	"entgo.io/ent"
+	"entgo.io/ent/dialect/entsql"
+	"entgo.io/ent/schema/field"
+	"entgo.io/ent/schema/index"
+
+	"github.com/gertd/go-pluralize"
+
+	"github.com/theopenlane/core/internal/ent/hooks"
+	"github.com/theopenlane/core/internal/ent/privacy/policy"
+	"github.com/theopenlane/core/internal/ent/validator"
+	"github.com/theopenlane/core/pkg/enums"
+)
+
+// CustomDomain holds the schema definition for the CustomDomain
+type CustomDomain struct {
+	SchemaFuncs
+
+	ent.Schema
+}
+
+const (
+	SchemaCustomDomain = "custom_domain"
+	maxDomainNameLen   = 255
+	maxTXTValueLen     = 64
+)
+
+func (CustomDomain) Name() string {
+	return SchemaCustomDomain
+}
+
+func (CustomDomain) GetType() any {
+	return CustomDomain.Type
+}
+
+func (CustomDomain) PluralName() string {
+	return pluralize.NewClient().Plural(SchemaCustomDomain)
+}
+
+// Fields of the CustomDomain
+func (CustomDomain) Fields() []ent.Field {
+	return []ent.Field{
+		field.String("cname_record").
+			Comment("the name of the custom domain").
+			Validate(validator.ValidateURL()).
+			MaxLen(maxDomainNameLen).
+			NotEmpty().
+			Immutable().
+			Annotations(
+				entgql.OrderField("cname_record"),
+			),
+		field.String("mappable_domain_id").
+			Comment("The mappable domain id that this custom domain maps to").
+			NotEmpty().
+			Immutable(),
+		field.String("txt_record_subdomain").
+			Comment("String to be prepended to the cname_record, used to evaluate domain ownership.").
+			Annotations(
+				entgql.Skip(entgql.SkipWhereInput, entgql.SkipMutationCreateInput, entgql.SkipMutationUpdateInput),
+			).
+			Annotations(
+				entgql.Skip(entgql.SkipMutationCreateInput, entgql.SkipMutationUpdateInput),
+			).
```

Comment:
Duplicate annotations, one should be removed
```suggestion
			Annotations(
				entgql.Skip(entgql.SkipWhereInput, entgql.SkipMutationCreateInput, entgql.SkipMutationUpdateInput),
			).
```